chap-cheri-riscv.tex

```diff
@@ -473,34 +456,26 @@ \subsubsection{Controlling Access to CSRs}
 \label{tab:risc-v-access-system-registers-whitelist}
 \end{table}
 
-\subsubsection{Capability Control and Status Registers (CCSRs)}
-\label{subsubsec-ccsrs}
-New per HART \xccsr{} \texttt{XLEN}-bit RISC-V CSRs are defined as per
-Figure~\ref{fig-ccsr} (shown for XLEN=32):
+\subsubsection{CHERI Extension Control}
 
-\begin{figure}[!h]
-\begin{center}
-\begin{bytefield}[bitwidth=\textwidth/34]{32}
-  \bitheader[endianness=big]{0,1,2,31} \\
-  \bitbox{31}{\textbf{WIRI}}
-  \bitbox{1}{\texttt{e}}
-\end{bytefield}
-\caption{\xccsr{} register format; WIRI bits are Write Ignore Read Ignore.}
-\label{fig-ccsr}
-\end{center}
-\end{figure}
+A new bit in the \menvcfg{} and \senvcfg{} CSRs is used to enable
+CHERI.  When CHERI is disabled, attempting to execute CHERI-specific
+instructions will raise an illegal instruction fault including loads
+and stores which use a capability register (excluding the implicit
+\DDC{} operand for legacy loads/stores) as the memory operand.
 
-\pmnote{WIRI has been removed from the RISC-V specification.  The spec
-  now typically uses WPRI (Write-Preserve, Read-Ignore) for such
-  undefined bits (e.g. in \texttt{mstatus}).  Both WARL (Write-Any,
-  Read-Legal) and WLRL (Write-Legal, Read-Legal) seem unsuitable here.
-  The RISC-V spec unsatisfyingly does not specify what happens if
-  software doesn't preserve values for WPRI, but then this spec
-  doesn't need to either.}
+Other CHERI extensions are always enabled regardless of the state of
+this bit.  Specifically, bounds and permissions on \PCC{} and \DDC{}
+are always honored.  Exceptions always copy \PCC{} to \xEPCC{} on
+exception entry and restore the full \PCC{} on exception return.
+Capability mode is always honored if enabled in \PCC{}.  Software
+which disables CHERI in lower modes must take care to ensure that
+\PCC{} and \DDC{} are set to suitable values while lower modes
+execute.
 
-\begin{description}
-\item [e] The \texttt{e} ``enable'' bit \ajnote{read only? / only wriFable in u-mode? only writable with a specific permission in pcc?} tells whether capability extensions are enabled or disabled.
-\end{description}
+Bit 8 in the \menvcfg{} and \senvcfg{} CSRs is defined as the CHERI
```

Comment:
My only concern is that 8 will conflict with the next allocated bit -- maybe we should use something like 0x1c which is right in the middle so should be safe for a while?

Reply:
Ok.